src/crystal/system/unix/file_descriptor.cr

```diff
@@ -131,14 +111,14 @@ module Crystal::System::FileDescriptor
     # Mark the handle open, since we had to have dup'd a live handle.
     @closed = false
 
-    evented_reopen
+    event_loop.close(self)
```

Comment:
I was going to propose to remove `IO::Evented#evented_reopen`, but then I realize this is part of the public API somehow. Maybe deprecate?

Reply:
Oh good point. Actually, all `evented_*` methods are documented API: https://crystal-lang.org/api/1.12.1/IO/Evented.html 🤦
I'm pretty sure this is totally unintended. Nobody should need to call these methods directly from user code.

Reply:
yes, this is what I mean with "public". I guess we can deprecate all of it...

Reply:
Eventually, everything from `IO::Evented` has to be removed, because it's event-loop specific implementation. These methods will need to be moved to the event loop implementation. Same for `IO::Overlapped` (which is thankfully nodoc).